src/ast/callback_visitor.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "ast.h"
+#include <functional>
+
+namespace bpftrace {
+namespace ast {
+
+using callback = std::function<void(Node *, void *)>;
```

Comment:
Absent a strong need to have C function pointers as callbacks, I'd suggest:
```
 using callback = std::function<void(Node *)>; 
```
and letting callers pass a lambda that captures whatever it needs by reference.

Reply:
ah, it looks good. will fix.